auth-plugin.c

```diff
@@ -91,10 +96,16 @@ int mosquitto_auth_unpwd_check(void *userdata, const char *username, const char
   GoString go_password = {password, strlen(password)};
   GoString go_clientid = {clientid, strlen(clientid)};
 
-  if(AuthUnpwdCheck(go_username, go_password, go_clientid)){
+  GoUint8 ret = AuthUnpwdCheck(go_username, go_password, go_clientid);
+
+  if(ret == AuthGranted){
```

Comment:
Yup, confirmed, this should be a switch according to what we agreed in the previous PR. Mind rebasing the branch so those changes are taken into account before I go on reviewing?

Reply:
It's better once rebased on master. The original PR want forked from PR #120 and I hopped that Github would not longer show diff of #120 once merged.
It wasn't the case and clearly made this PR unreviewable.